### PR TITLE
Feature/modular app templates

### PR DESCRIPTION
**Pull request by PierreJeanjacquot**
# modular template

choose init type (`hello-world` or `advanced`)

image
image

- `hello-world` - get started with a simple "hello-world" app (only use `args`)

image

- `advanced` - pick the features you want
  - args
  - protected data
  - input files
  - requester secrets
  - app secret

image

User choices produce a template with selected features (technically we strip unwanted features from a "full-featured" template)

## misc

- protectedData mock is now stored in `mock/protectedData` and copied in `input` folder when needed 
- `input` is now cleaned before each `iapp test` 
- `iapp.config.json` has been added to ignore files as it may contain some secrets

### CODE REVIEW

File: cli/src/cmd/init.js [hunk 1]
```diff
@@ -33,22 +33,80 @@ export async function init() {
...
+            type: 'confirm',
+            name: 'useArgs',
+            message: 'Would you like to use positional args inside your iApp?',
+            initial: true,
+          },
+          {
+            type: 'confirm',
+            name: 'useProtectedData',
+            message:
+              'Would you like to access a protected data inside your iApp?',
```
Comment (PierreJeanjacquot) [inline cli/src/cmd/init.js:80]: looking for suggestions to improve understandability

File: cli/src/cmd/init.js [hunk 2]
```diff
@@ -33,22 +33,80 @@ export async function init() {
...
+            initial: false,
+          },
+          {
+            type: 'confirm',
+            name: 'useInputFile',
+            message:
+              'Would you like to use input files inside your iApp? (input files are files downloaded from the internet)',
```
Comment (PierreJeanjacquot) [inline cli/src/cmd/init.js:87]: looking for suggestions to improve understandability

File: cli/src/cmd/init.js [hunk 3]
```diff
@@ -33,22 +33,80 @@ export async function init() {
...
+            initial: false,
+          },
+          {
+            type: 'confirm',
+            name: 'useRequesterSecret',
+            message:
+              'Would you like to use requester secrets inside your iApp? (requester secrets are secrets provisioned by the user)',
```
Comment (PierreJeanjacquot) [inline cli/src/cmd/init.js:94]: looking for suggestions to improve understandability

File: cli/src/cmd/init.js [hunk 4]
```diff
@@ -33,22 +33,80 @@ export async function init() {
...
+            initial: false,
+          },
+          {
+            type: 'confirm',
+            name: 'useAppSecret',
+            message:
+              'Would you like to use an app secret inside your iApp? (app secret is a secret provisioned by the app owner)',
```
Comment (PierreJeanjacquot) [inline cli/src/cmd/init.js:101]: looking for suggestions to improve understandability

File: cli/src/cmd/init.js (hunk 2 above)
Comment (Le-Caignec) [reply cli/src/cmd/init.js:87]: `Do you want to use input files in your iApp? (Note: Input files are typically downloaded from the internet)`

File: cli/src/cmd/init.js (hunk 1 above)
Comment (Le-Caignec) [reply cli/src/cmd/init.js:80]: `Do you want to access protected data within your iApp?`

File: cli/src/cmd/init.js (hunk 3 above)
Comment (Le-Caignec) [inline cli/src/cmd/init.js:94]:
```suggestion
              'Would you like to use requester secrets inside your iApp? (requester secrets are secrets that will be provided by any users that will run your iapp)',
```

File: cli/src/cmd/init.js (hunk 4 above)
Comment (Le-Caignec) [inline cli/src/cmd/init.js:101]:
```suggestion
              'Would you like to use an app secret inside your iApp? (app secret is a secret provisioned by the app owner linked to your iapp that will never changed)',
```

File: cli/src/cmd/init.js [hunk 5]
```diff
@@ -33,22 +33,80 @@ export async function init() {
...
 
+    const INIT_BASIC = 'basic';
+    const INIT_ADVANCED = 'advanced';
+    const { initType } = await spinner.prompt({
+      type: 'select',
+      name: 'initType',
+      message: 'What kind of project do you want to init?',
+      choices: [
+        {
+          title: 'hello-world',
```
Comment (Le-Caignec) [inline cli/src/cmd/init.js:49]: what do you think about using PascalCase here ? 
```suggestion
          title: 'HelloWorld',
```

**Review by Le-Caignec — COMMENTED** [review]
Good work 💪🏻💪🏻

File: cli/src/cmd/init.js (hunk 5 above)
Comment (PierreJeanjacquot) [reply cli/src/cmd/init.js:49]: plain English seems even better :)

```suggestion
          title: 'Hello World',
```

File: cli/src/cmd/init.js [hunk 6]
```diff
@@ -33,22 +33,80 @@ export async function init() {
...
+            name: 'useInputFile',
+            message:
+              'Would you like to use input files inside your iApp? (input files are files downloaded from the internet)',
+            initial: false,
+          },
+          {
+            type: 'confirm',
+            name: 'useRequesterSecret',
+            message:
+              'Would you like to use requester secrets inside your iApp? (requester secrets are secrets that will be provided by any users that will run your iapp)',
```
Comment (PierreJeanjacquot) [inline cli/src/cmd/init.js:94]:
```suggestion
              'Would you like to use requester secrets inside your iApp? (requester secrets are secrets that will be provided by any users that will run your iApp)',
```

File: cli/src/cmd/init.js [hunk 7]
```diff
@@ -33,22 +33,80 @@ export async function init() {
...
+            initial: false,
+          },
+          {
+            type: 'confirm',
+            name: 'useAppSecret',
+            message:
+              'Would you like to use an app secret inside your iApp? (app secret is an immutable secret provisioned once by the iapp owner)',
```
Comment (PierreJeanjacquot) [inline cli/src/cmd/init.js:101]:
```suggestion
              'Would you like to use an app secret inside your iApp? (app secret is an immutable secret provisioned once by the iApp owner)',
```

**Pull request merged**